core/src/mender-client.c

```diff
@@ -990,7 +990,10 @@ mender_client_update_work_function(void) {
 
             case MENDER_UPDATE_STATE_INSTALL:
                 mender_log_info("Download done, installing artifact");
-                mender_client_publish_deployment_status(deployment_id, MENDER_DEPLOYMENT_STATUS_INSTALLING);
+                if (MENDER_ABORTED == mender_client_publish_deployment_status(deployment_id, MENDER_DEPLOYMENT_STATUS_INSTALLING)) {
+                    update_state = MENDER_UPDATE_STATE_FAILURE;
+                    continue;
```

Comment:
We need to also save the state, etc., right? Let's just assign `ret` here and check it also before calling the update module's callback below so that the code skips everything until `NEXT_STATE` if `ret != MENDER_OK`.